Merge 2.18 code freeze

mokagio opened the pull request:
This includes a few changes that were necessary to build the app in macOS versions without Python 2.7, and with the Xcode 14.3 toolchain.

## Next steps

I'll need to open a PR to track upgrading the old Python localization script to Python 3 (via built-in `2to3`) and show that the diff in the `strings` it generates is the same as this PR.

The `strings` here were generated on an older Mac on which `python` was still available, but I think it would be foolish to expect an old Mac to keep being available. 

Worth pointing out that we could upgrade the tooling instead, but give the shift in focus away from Simplenote macOS, I think the 10 minutes it'll take me to verify the diff and make a PR will be sufficient.

mokagio committed:
Bump version number

mokagio committed:
Release Notes: add new section for next version (2.19)

mokagio committed:
Update strings for localization

mokagio committed:
Update release notes for version 2.18

mokagio committed:
Update metadata strings

mokagio committed:
Update CocoaPods to latest version, 1.10.2

This was necessary to address a failure running `bundle exec pod
install`:

> Updating local specs repositories
> [!] CDN: trunk URL couldn't be downloaded: https://cdn.cocoapods.org/deprecated_podspecs.txt Response: URL using bad/illegal format or missing URL

mokagio committed:
Make pods use application's macOS deployment target

Without this, some of the old pods in this app would fail to build in
Xcode 14.3

mokagio committed:
Track `Package.resolved` upgrade to version 2 done by Xcode 14.3

mokagio committed:
pod deployment targ

mokagio committed:
Run Sparkle's `codesign_embedded_executable` with `python3`

macOS no longer ships `python`. I run the script through the `2to3`
official tool, https://docs.python.org/3/library/2to3.html and no source
change was required.

mokagio committed:
Update CocoaPods to latest version, 1.12.0

This was done to see if it helped with the Simplenote target failing to
Archive during the "[CP] Embed Pods Framework" phase.

Full error:

```
PhaseScriptExecution [CP]\ Embed\ Pods\ Frameworks /Users/gio/Developer/a8c/snmacos/DerivedData/Simplenote/Build/Intermediates.noindex/ArchiveIntermediates/Simplenote/IntermediateBuildFilesPath/Simplenote.build/Release/Simplenote.build/Script-BCBFB16555965D73674605BA.sh (in target 'Simplenote' from project 'Simplenote')
    cd /Users/gio/Developer/a8c/snmacos
    /bin/sh -c /Users/gio/Developer/a8c/snmacos/DerivedData/Simplenote/Build/Intermediates.noindex/ArchiveIntermediates/Simplenote/IntermediateBuildFilesPath/Simplenote.build/Release/Simplenote.build/Script-BCBFB16555965D73674605BA.sh

mkdir -p /Users/gio/Developer/a8c/snmacos/DerivedData/Simplenote/Build/Intermediates.noindex/ArchiveIntermediates/Simplenote/BuildProductsPath/Release/Simplenote.app/Contents/Frameworks
Symlinked...
rsync --delete -av --filter P .*.?????? --links --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" --filter "- Headers" --filter "- PrivateHeaders" --filter "- Modules" "../../../IntermediateBuildFilesPath/UninstalledProducts/macosx/AutomatticTracks.framework" "/Users/gio/Developer/a8c/snmacos/DerivedData/Simplenote/Build/Intermediates.noindex/ArchiveIntermediates/Simplenote/InstallationBuildProductsLocation/Applications/Simplenote.app/Contents/Frameworks"
building file list ... rsync: link_stat "/Users/gio/Developer/a8c/snmacos/../../../IntermediateBuildFilesPath/UninstalledProducts/macosx/AutomatticTracks.framework" failed: No such file or directory (2)
done

sent 29 bytes  received 20 bytes  98.00 bytes/sec
total size is 0  speedup is 0.00
rsync error: some files could not be transferred (code 23) at /AppleInternal/Library/BuildRoots/810eba08-405a-11ed-86e9-6af958a02716/Library/Caches/com.apple.xbs/Sources/rsync/rsync/main.c(996) [sender=2.6.9]
Command PhaseScriptExecution failed with a nonzero exit code
```

mokagio committed:
Add Pods `post_install` workaround for Xcode 14.3 issue

See previous commit and
https://github.com/CocoaPods/CocoaPods/issues/11808

mokagio commented on External/Sparkle/bin/codesign_embedded_executable:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
It's not entirely appropriate to upgrade this file since it should be a copy of Sparkle, but I think we can close an eye given this is the only change that was required to use Python 3

mokagio commented on Gemfile:
```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'cocoapods', '>= 1.10'
+gem 'cocoapods', '~> 1.12'
```
I had to upgrade CocoaPods because I run into a CDN issue with that older version.

```
[!] CDN: trunk URL couldn't be downloaded: https://cdn.cocoapods.org/deprecated_podspecs.txt Response: URL using bad/illegal format or missing URL
```

mokagio commented on Simplenote.xcworkspace/xcshareddata/swiftpm/Package.resolved:
```diff
@@ -1,34 +1,32 @@
+      }
+    }
+  ],
+  "version" : 2
```
I don't remember at which point, but Xcode changed the `.resolved` version in the last couple of years or so. This is just what happened as I opened the app in Xcode 14.3. You can see the packages haven't changed version.

crazytonyli commented on Podfile:
```diff
@@ -23,3 +25,34 @@ abstract_target 'Automattic' do
+      end
+    end
+  end
+end
```
I just noticed CocoaPods [released 1.12.1](https://github.com/CocoaPods/CocoaPods/releases/tag/1.12.1) 4 hours ago, have you tried the latest version which hopefully fixed these integration issues in Xcode 14.3?

mokagio replied:
Neat! I'll try it out, it does mention the fix we need:

<img width="754" alt="Screenshot 2023-04-19 at 3 55 56 pm" src="https://user-images.githubusercontent.com/1218433/232979777-ef69f7e6-1e9d-4fec-aae0-9742880c1173.png">

mokagio committed:
Update to freshly shipped CocoaPods 1.12.1 and remove 14.3 workaround

Hat tip @crazytonyli

See
https://github.com/Automattic/AnalyticsSDK-iOS/pull/12#discussion_r1169289423
